From 6a26c187beb14ccdee95d95ab0194410ba9e0ded Mon Sep 17 00:00:00 2001
From: Aloka Dixit <quic_alokad@quicinc.com>
Date: Tue, 20 Jul 2021 13:50:06 -0700
Subject: [PATCH 21/23] nl80211: check driver capabilities for beacon rates

Set flag WPA_DRIVER_FLAGS2_BEACON_RATE_EHT if the driver advertises
support for EHT rates for beacon transmission.

Signed-off-by: Aloka Dixit <quic_alokad@quicinc.com>
---
 src/drivers/driver.h              | 2 ++
 src/drivers/driver_nl80211_capa.c | 4 ++++
 2 files changed, 6 insertions(+)

--- a/src/drivers/driver.h
+++ b/src/drivers/driver.h
@@ -2202,6 +2202,7 @@ struct wpa_driver_capa {
  * frames in STA mode
  */
 #define WPA_DRIVER_FLAGS2_PROT_RANGE_NEG_STA	0x0000000000002000ULL
+#define WPA_DRIVER_FLAGS2_BEACON_RATE_EHT       0x0000000000004000ULL
 /** Driver supports MLO in station/AP mode */
 #define WPA_DRIVER_FLAGS2_MLO			0x0000000000004000ULL
 	u64 flags2;
--- a/src/drivers/driver_nl80211_capa.c
+++ b/src/drivers/driver_nl80211_capa.c
@@ -570,6 +570,10 @@ static void wiphy_info_ext_feature_flags
 		capa->flags2 |= WPA_DRIVER_FLAGS2_BEACON_RATE_HE;
 
 	if (ext_feature_isset(ext_features, len,
+			      NL80211_EXT_FEATURE_BEACON_RATE_EHT))
+		capa->flags2 |= WPA_DRIVER_FLAGS2_BEACON_RATE_EHT;
+
+	if (ext_feature_isset(ext_features, len,
 			      NL80211_EXT_FEATURE_SET_SCAN_DWELL))
 		capa->rrm_flags |= WPA_DRIVER_FLAGS_SUPPORT_SET_SCAN_DWELL;
 
